src/components/Inputs/Inputs.js

```diff
@@ -2,141 +2,157 @@ import React, { Component } from 'react';
 
 import { View,
   StyleSheet,
-  Image
+  Image,
+  Text
 } from 'react-native';
 
+import inputImages from '../../img/inputs/index';
+
 export default class Inputs extends Component {
   constructor() {
     super();
     this.state = {};
   }
 
   renderInputs = (inputs) => {
-    // TODO: figure out the best way to implement keys here so RN will stop bitching
     return inputs.map((input, index) => {
       switch (input) {
-        // TODO: add image sources here
+        // TODO: fix up missing image sources
         case '1':
           return (
```

Comment:
Recap of discussion:
dnguyen [11:32 AM] 
We still would only have one file requiring the images, it would just be the sourcemap config
this is the scenario in my head

say you have `d, df, 4` => `[d, df, 4]`. You already got a `inputs.map( (input) => {} )` going. 
For each input, return src = `inputImages[ input ]`
The inputImages object would look like `{ '<key>' : <img src>, ... }`

This eliminates the large switch case in the component, as it would only be parsing through the input notatinos and pulling the appropriate images for it
You would still be returning individual `<image/>` components for each unique notation part of the whole move notation needed

If the data source changes notation, it wouldn’t matter since we would still just be pulling the image for whatever notation is needed.

sjones [11:40 AM] 
that makes sense. Would the image map file match its own key based on the parameter we give it?

dnguyen [11:41 AM] 
Nah. The imageMap is just a simple dictionary we use to map any individual unique notation part of a moves list that we need.
It would have a key for `2`, or `2+3`, etc (since those both have their own unique images)

Unless you want to go further and do the parsing all in the imageMap and return an array of images and text. That’s a lot of work for it, so I would hold off on doing that